src/hb-ot-var.cc

```diff
@@ -77,9 +82,17 @@ hb_ot_var_get_axis_count (hb_face_t *face)
 
 /**
  * hb_ot_var_get_axes:
+ * @face: #hb_face_t to work upon
+ * @start_offset: offset of the first lookup to retrieve
+ * @axes_count: (inout) (allow-none): Input = the maximum number of variation axes to return;
+ *                Output = the actual number of variation axes returned (may be zero)
+ * @axes_array: (out) (array length=axes_count): The array of variation axes found
```

Comment:
Shouldn’t it be `(out caller-allocates)`?

Reply:
Honestly: I'm not sure. You have the experience with GObject Introspection etc that I don't, so I trust your judgement on it.

Reply:
May memory is fading, so whatever is done elsewhere and lets see if anyone complains.

Reply:
@behdad Any quick answer there?

Reply:
> Shouldn’t it be `(out caller-allocates)`?

That's what we do in other places it seems, yes.  Though, by far not in all places.  I think it should be added to many more places.

Reply:
Probably all of these need fixing:
```
$ git grep '(out).*(array length='
hb-buffer-serialize.cc: * @buf: (out) (array length=buf_size) (element-type uint8_t): output string to
hb-ot-color.cc: * @colors: (out) (array length=color_count) (nullable): The array of #hb_color_t records found
hb-ot-color.cc: * @layers: (out) (array length=layer_count) (nullable): The array of layers found
hb-ot-layout.cc: * @point_array: (out) (array length=point_count): The array of attachment points found for the query
hb-ot-layout.cc: * @caret_array: (out) (array length=caret_count): The array of caret positions found for the query
hb-ot-layout.cc: * @script_tags: (out) (array length=script_count): The array of #hb_tag_t script tags found for the query
hb-ot-layout.cc: * @feature_tags: (out) (array length=feature_count): Array of feature tags found in the table
hb-ot-layout.cc: * @language_tags: (out) (array length=language_count): Array of language tags found in the table
hb-ot-layout.cc: * @feature_indexes: (out) (array length=feature_count): The array of feature indexes found for the query
hb-ot-layout.cc: * @feature_tags: (out) (array length=feature_count): The array of #hb_tag_t feature tags found for the query
hb-ot-layout.cc: * @lookup_indexes: (out) (array length=lookup_count): The array of lookup indexes found for the query
hb-ot-layout.cc: * @lookup_indexes: (out) (array length=lookup_count): The array of lookups found for the query
hb-ot-math.cc: * @variants: (out) (array length=variants_count): array of variants returned
hb-ot-math.cc: * @parts: (out) (array length=parts_count): the glyph parts returned
hb-ot-name.cc: * Returns: (out) (transfer none) (array length=num_entries): Array of available name entries.
```
except for the last one I think.